### PR TITLE
Add a way to filter the health overview logs

### PR DESCRIPTION
**Pull request by gmarouli**
The health periodic logger exposes through the logs an overview of the status of each health indicator. In order to be able to filter effectively and count this log lines, we would like to add an extra flag. The json will look like this:

```
"elasticsearch": {
      "health_overview": true,
      "health": {
        "disk": {
          "status": "green"
        },
        "shards_availability": {
          "status": "green"
        },
        "repository_integrity": {
          "status": "green"
        },
        "slm": {
          "status": "green"
        },
        "master_is_stable": {
          "status": "green"
        },
        "ilm": {
          "status": "green"
        },
        "shards_capacity": {
          "status": "green"
        },
        "overall": {
          "status": "green"
        }
      }
    }
```

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java [hunk 1]
```diff
@@ -189,18 +189,20 @@ static Map<String, Object> convertToLoggedFields(List<HealthIndicatorResult> ind
         }
 
         final Map<String, Object> result = new HashMap<>();
+        // This flag is used to filter this log lines for analysis
```
Comment (n1v0lg) [inline server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java:192]: Nit:
```suggestion
        // This flag is used to filter this log line for analysis
```

**Review by n1v0lg — APPROVED** [review]
LGTM. Just missing a `>non-issue` label. 

One optional consideration: we could include the flag in the `health` object instead, i.e., `{"health": {"overview": true, ...}}`. I'm not sure if we could alternatively just use the existence of `health.status` as a indicator in our filtering -- all good with the extra field if that makes things easier.

~Alternatively, wondering if we can simply check for~

**Comment by n1v0lg** [comment]
> Alternatively, wondering if we can simply check for

Ah sorry this was just an editting whoopla -- nothing else in mind! My (slight) preference would be a nested field `"health.overview"` but either way works!

**Comment by mattc58** [comment]
So @gmarouli this needs to exist because we do not think that Kibana Alerts support `exists` type queries? I know for the dashboards we're able to use something like `elasticsearch.health.overall.status:*` and it gets us this desired behavior.

**Comment by gmarouli** [comment]
> So @gmarouli this needs to exist because we do not think that Kibana Alerts support `exists` type queries? I know for the dashboards we're able to use something like `elasticsearch.health.overall.status:*` and it gets us this desired behavior.

Let me try that!

**Comment by gmarouli** [comment]
> So @gmarouli this needs to exist because we do not think that Kibana Alerts support `exists` type queries? I know for the dashboards we're able to use something like `elasticsearch.health.overall.status:*` and it gets us this desired behavior.

Unfortunately this doesn't work, I believe it's not accepting the wildcard but matching it as a keyword.

**Comment by n1v0lg** [comment]
Yee there's an open ER for it: https://github.com/elastic/enhancements/issues/13823

**Pull request closed without merging**